language/tools/move-mv-llvm-compiler/src/stackless/rttydesc.rs

```diff
@@ -0,0 +1,209 @@
+        &[type_name_ty, type_descrim_ty, type_info_ptr_ty]
+    };
+
+    td_llty.set_struct_body(field_tys);
+}
+
+pub fn define_llvm_tydesc(
+    llcx: &llvm::Context,
+    llmod: &llvm::Module,
+    mty: &mty::Type,
+) -> llvm::Global {
+    let name = global_tydesc_name(mty);
+    match llmod.get_global(&name) {
+        Some(g) => g,
+        None => {
+            let ll_tydesc_ty = get_llvm_tydesc_type(llcx);
+            let ll_tydesc_ty = ll_tydesc_ty.as_any_type();
+            let ll_global = llmod.add_global(ll_tydesc_ty, &name);
+            ll_global.set_constant();
+            let ll_constant = tydesc_constant(llcx, llmod, mty);
+            let ll_constant_ty = ll_constant.llvm_type();
+            ll_global.set_initializer(ll_constant);
+            ll_global
+        }
+    }
+}
+
+fn tydesc_constant(llcx: &llvm::Context, llmod: &llvm::Module, mty: &mty::Type) -> llvm::Constant {
+    let ll_const_type_name = type_name_constant(llcx, llmod, mty);
+    let ll_const_type_descrim = {
+        let ll_ty = llcx.int32_type();
+        llvm::Constant::int(ll_ty, type_descrim(mty))
+    };
+    let ll_const_type_info_ptr = {
+        let ll_global_type_info = define_type_info_global(llcx, llmod, mty);
+        ll_global_type_info.ptr()
+    };
+    let ll_const = llcx.const_named_struct(
+        &[
+            ll_const_type_name,
+            ll_const_type_descrim,
+            ll_const_type_info_ptr,
+        ],
+        TD_NAME,
+    );
+    ll_const
+}
+
+fn type_name_constant(
+    llcx: &llvm::Context,
+    llmod: &llvm::Module,
+    mty: &mty::Type,
+) -> llvm::Constant {
+    let name = type_name(mty);
+    let len = name.len();
+
+    // Create a static string and take a constant pointer to it.
+    let ll_static_bytes_ptr = {
+        let global_name = global_tydesc_name_name(mty);
+        match llmod.get_global(&global_name) {
+            Some(g) => g.ptr(),
+            None => {
+                let ll_const_string = llcx.const_string(&name);
+                let ll_array_ty = ll_const_string.llvm_type();
+                let ll_global = llmod.add_global(ll_array_ty, &global_name);
+                ll_global.set_constant();
+                ll_global.set_initializer(ll_const_string.as_const());
+                ll_global.ptr()
+            }
+        }
+    };
+
+    let ll_ty_u64 = llcx.int64_type();
+    let ll_const_len = llvm::Constant::int(ll_ty_u64, len as u64);
+
+    let ll_const_struct = llcx.const_struct(&[ll_static_bytes_ptr, ll_const_len]);
+
+    ll_const_struct
+}
+
+fn type_name(mty: &mty::Type) -> String {
+    use mty::{PrimitiveType, Type};
+    let name = match mty {
+        Type::Primitive(PrimitiveType::U64) => "u64",
+        _ => todo!(),
+    };
+
+    format!("{name}")
+}
+
+/// The values here correspond to `move_native::rt_types::TypeDesc`.
+fn type_descrim(mty: &mty::Type) -> u64 {
+    use mty::{PrimitiveType, Type};
+    match mty {
+        Type::Primitive(PrimitiveType::U64) => 3,
```

Comment:
Nit: Instead of hardcoding '3' (and the other cases that will be there in the future), can we reference the 'enum TypeDesc ' directly (or otherwise share it)?

Reply:
I've changed move-native to export TypeDesc and linked to it from move-mv-llvm-compiler and used the definition here.